lib/vorbisfile.c

```diff
@@ -1989,6 +1990,8 @@ long ov_read_filter(OggVorbis_File *vf,char *buffer,int length,
     long channels=ov_info(vf,-1)->channels;
     long bytespersample=word * channels;
     vorbis_fpu_control fpu;
+
+    if(!channels)return(OV_EINVAL);
```

Comment:
Likewise, if we're being defensive, maybe borrow the check from `vorbis_encode_setup_init()`?
```c
if(channels<1||channels>255)return(OV_EINVAL);
```